apps/builder/app/builder/shared/tree/use-editable.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  useRef,
+  useEffect,
+  type KeyboardEventHandler,
+  type KeyboardEvent,
+  type FocusEvent,
+} from "react";
+
+export const useEditable = ({
```

Comment:
I just realized you used content editable .... why risking a lot of bugs instead of just using a plain input?

So much that can happen in editable, its a nightmare

Reply:
I had the same idea first, but then i looked at how tokens are being edited in the right side panel with contenteditable. And thought that's the preferred way. And changed it to use `contenteditable` 😅

Reply:
actually lets check with @TrySound maybe that's indeed better? I remember using contenteditable there because there was some issue, maybe because you can't use input inside a button and I didn't want change the rendering? 

I don't remember ...

Reply:
I guess it's fine